### PR TITLE
feature: Account verification page B&F

### PR DESCRIPTION
**Pull request by ProggerOleg**
## Describe your changes

- Created layout and logic of account verification task
- Added input validation(number of digits, is verification code correct )
- Created custom answers depending on whether user seeker or caregiver
- Created request to backend on submit
- Added RTK query for request new verification code, implemented it's logic into component

Screenshot from 2023-11-10 19-26-32

Screenshot from 2023-11-13 16-31-57

Screenshot from 2023-11-13 16-32-30

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://homecareaid.atlassian.net/jira/software/projects/AC/boards/1?selectedIssue=AC-8)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/components/sendOTP/OTPMessageField.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+const OTPMessageField: React.FC<OTPMessageFieldProps> = ({ onSubmit }): JSX.Element => {
+  const { t } = useTranslation();
+  const expectedCode = process.env.NEXT_PUBLIC_OTP_CODE_EXAMPLE;
+
+  const [code, setCode] = useState(['', '', '', '']);
+
+  const verificationCode = code.join('');
+  const codeDoesNotMatch = verificationCode.length > 0 && verificationCode.length <= 4 && verificationCode !== expectedCode;
```
Comment (lyudad) [inline src/components/sendOTP/OTPMessageField.tsx:24]: https://www.notion.so/Git-rules-849f8ca8b3db4ded926d762c6fff8798 1.1.4

File: src/components/sendOTP/SuccessfulVerification.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        <SuccessAccountVerificationContainer>
+                <IconContainer>
+                    <SuccessfulValidationIcon />
+                </IconContainer>
+               <TextBlock>
+               <StyledParagraphSuccess>
+                {t("Successfully_Verified")}
+                </StyledParagraphSuccess>
+                <p>
+                    {profile === "Seeker" ? t("Seeker.Successfully_Verified_Text") : t("Caregiver.Successfully_Verified_Text")}
```
Comment (lyudad) [inline src/components/sendOTP/SuccessfulVerification.tsx:26]: https://www.notion.so/Git-rules-849f8ca8b3db4ded926d762c6fff8798 1.1.5

File: src/components/sendOTP/SuccessfulVerification.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                </p>
+               </TextBlock>
+
+                <SubmitButtonContainer>
+                    <FilledButton
+                        fullWidth>
+                    {profile === "Seeker" ? t("Seeker.Successfully_Verified_Button") : t("Caregiver.Successfully_Verified_Button")}
```
Comment (lyudad) [inline src/components/sendOTP/SuccessfulVerification.tsx:33]: same

File: src/components/sendOTP/index.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next';
+
+import SignUpHeader from 'src/components/reusable/header';
+import OTPMessageField from 'src/components/sendOTP/OTPMessageField';
+import SuccessfulVerification from 'src/components/sendOTP/SuccessfulVerification';
+
+const AccountVerification = () :JSX.Element => {
+    const { t } = useTranslation();
+    const [isSubmitted, setIsSubmitted] = useState(false);
```
Comment (lyudad) [inline src/components/sendOTP/index.tsx:10]: type

File: src/components/sendOTP/index.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import { useTranslation } from 'react-i18next';
+
+import SignUpHeader from 'src/components/reusable/header';
+import OTPMessageField from 'src/components/sendOTP/OTPMessageField';
+import SuccessfulVerification from 'src/components/sendOTP/SuccessfulVerification';
+
+const AccountVerification = () :JSX.Element => {
+    const { t } = useTranslation();
+    const [isSubmitted, setIsSubmitted] = useState(false); 
+    const profile = process.env.NEXT_PUBLIC_PROFILE || '';
```
Comment (lyudad) [inline src/components/sendOTP/index.tsx:11]: it will be reinialise with each rerender

File: src/pages/account-verification/index.tsx [hunk 6]
```diff
@@ -0,0 +1 @@
+export { default } from '../../components/sendOTP/index'
```
Comment (lyudad) [inline src/pages/account-verification/index.tsx:1]: https://www.notion.so/Git-rules-849f8ca8b3db4ded926d762c6fff8798 1.3.5

File: src/components/sendOTP/OTPMessageField.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+interface OTPMessageFieldProps {
+  onSubmit: () => void;
+}
+
+
+const OTPMessageField: React.FC<OTPMessageFieldProps> = ({ onSubmit }): JSX.Element => {
+  const { t } = useTranslation();
+  const expectedCode = process.env.NEXT_PUBLIC_OTP_CODE_EXAMPLE;
+
+  const [code, setCode] = useState(['', '', '', '']);
```
[the remark this answers is not in the document]
Comment (ProggerOleg) [reply src/components/sendOTP/OTPMessageField.tsx:21]: I have added type,cant understand what's wrong with this

File: src/components/sendOTP/index.tsx (hunk 4 above)
Comment (ProggerOleg) [reply src/components/sendOTP/index.tsx:14]: Screenshot from 2023-11-15 21-18-44
I don't understand what's wrong, for me it looks like I've got all needed types, provide me with extra info please

File: src/components/sendOTP/OTPMessageField.tsx (hunk 1 above)
Comment (ProggerOleg) [reply src/components/sendOTP/OTPMessageField.tsx]: done

File: src/components/sendOTP/SuccessfulVerification.tsx (hunk 2 above)
Comment (ProggerOleg) [reply src/components/sendOTP/SuccessfulVerification.tsx]: done

File: src/components/sendOTP/SuccessfulVerification.tsx (hunk 3 above)
Comment (ProggerOleg) [reply src/components/sendOTP/SuccessfulVerification.tsx]: done

File: src/components/sendOTP/index.tsx (hunk 4 above)
Comment (ProggerOleg) [reply src/components/sendOTP/index.tsx]: done,  sorry for inconvenience

File: src/components/sendOTP/index.tsx (hunk 5 above)
Comment (ProggerOleg) [reply src/components/sendOTP/index.tsx]: fixed

File: src/pages/account-verification/index.tsx (hunk 6 above)
Comment (ProggerOleg) [reply src/pages/account-verification/index.tsx]: fixed

File: src/components/sendOTP/OTPMessageField.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (ProggerOleg) [reply src/components/sendOTP/OTPMessageField.tsx]: done

File: src/components/sendOTP/SuccessfulVerification.tsx [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+
+const SuccessfulVerification = ({ profile }:  {profile: string}): JSX.Element  => {
+
+    const { t } = useTranslation();
+
+    const roles = {
+        SEEKER: 'Seeker',
+        CAREGIVER: 'Caregiver'
+      };
```
[the remark this answers is not in the document]
Comment (ProggerOleg) [reply src/components/sendOTP/SuccessfulVerification.tsx]: fixed

File: src/components/sendOTP/index.tsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+import SuccessfulVerification from 'src/components/sendOTP/SuccessfulVerification';
+
+interface AccountVerificationProps {
+  onSubmit: () => void;
+}
+
+const AccountVerification: React.FC<AccountVerificationProps> = () :JSX.Element => {
+    const { t } = useTranslation();
+    const [isSubmitted, setIsSubmitted] = useState<boolean>(false); 
+    const profile :string = useMemo(() => process.env.NEXT_PUBLIC_PROFILE || '', []);
```
Comment (lyudad) [inline src/components/sendOTP/index.tsx:15]: зачем эта константа в компоненте?

File: src/components/sendOTP/styles.ts [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+`;
+
+export const IconContainer = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 10em; 
+  height: 10em;
+  border-radius: 50%;
+  background-color: #08BCB81F;
```
Comment (lyudad) [inline src/components/sendOTP/styles.ts:34]: та йо-ма-йо, ПОЖАЛУСТА проверь свой код

File: src/components/sendOTP/AccountVerification.tsx [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+import useVerification from 'src/hooks/useVerification';
+import { AccountVerificationContainer, IconContainer, StyledParagraph, StyledParagraphMain, SubmitButtonContainer } from 'src/components/sendOTP/styles';
+
+
+interface OTPMessageFieldProps { 
+  onSubmit: () => void;
+}
+const OTPMessageField: React.FC<OTPMessageFieldProps> = ({ onSubmit }): JSX.Element => {
+  const { t } = useTranslation();
+  const { code, codeDoesNotMatch, handleInputChange, handleSubmit, fetchNewCode } = useVerification({onSubmit});
```
Comment (lyudad) [inline src/components/sendOTP/AccountVerification.tsx:16]: что то с притиром не так, туд должна быть пустая строка

File: src/components/sendOTP/AccountVerification.tsx [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+import { FilledButton } from 'src/components/reusable/FilledButton';
+import DigitTextField from 'src/components/sendOTP/DigitInputField';
+import useVerification from 'src/hooks/useVerification';
+import { AccountVerificationContainer, IconContainer, StyledParagraph, StyledParagraphMain, SubmitButtonContainer } from 'src/components/sendOTP/styles';
+
+
+interface OTPMessageFieldProps { 
+  onSubmit: () => void;
+}
+const OTPMessageField: React.FC<OTPMessageFieldProps> = ({ onSubmit }): JSX.Element => {
```
Comment (lyudad) [inline src/components/sendOTP/AccountVerification.tsx:14]: ту тоже нет пустой строки, проверь настройки еслинта и притира

File: src/components/sendOTP/styles.ts [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+  line-height: 1.8;
+  width: 300px;
+`;
+
+export const StyledParagraphMain = styled.a`
+  font-weight: ${ typography.fontWeightMedium };
+  color: ${ PRIMARY.navy };
+  text-decoration:none;
+  text-align: center;
+  line-height: 1.8;
```
Comment (lyudad) [inline src/components/sendOTP/styles.ts:60]: рекомендую в будущем найти ка верстать без этого, щаще всего это признак плохой верстки

File: src/hooks/useVerification.ts [hunk 14]
```diff
@@ -0,0 +1,98 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useRouter } from 'next/router';
+
+import { RootState } from 'src/redux/store';
+import { ACCOUNT_VERIFIED_RESPONSE_MESSAGE, NO_ACCOUNT_TO_VERIFY_RESPONSE_MESSAGE } from 'src/components/sendOTP/constants';
+import { useRequestNewVerificationCodeMutation, useSubmitVerificationCodeMutation } from 'src/redux/api/accountVerificationAPI';
+
+import jwt_decode from 'jwt-decode';
```
Comment (lyudad) [inline src/hooks/useVerification.ts:9]: https://www.notion.so/Git-rules-849f8ca8b3db4ded926d762c6fff8798 1.1.14

**Pull request merged**